### ISSUE TITLE
Make AppThemeBinding type public

### ISSUE THREAD

**Issue by egvijayanand** [dotnet/maui#19725]
### Description

The type `AppThemeBinding` is the equivalent of `AppThemeBindingExtension` when working with C# Markup.

But this type is marked as internal. So, unable to make use of it while defining theme-based Resource definition (as an instance assigned to the value of the Setter).

_For inline View definition, there's an extension method available. But not for Setter as there's no instance available to invoke the extension method. Refer to the intended use case for the sample._

### Public API Changes

Make the **AppThemeBinding** type public.

https://github.com/dotnet/maui/blob/c3221bf08d291c9f69070aa11d6d60f61a569aae/src/Controls/src/Core/AppThemeBinding.cs#L9C1-L9C37

### Intended Use-Case

Allows resource reuse.

```cs
Resources.Add("MauiLabel", new Style(typeof(Label))
{
    Setters =
    {
        new() { Property = Label.TextColorProperty, Value = new AppThemeBinding() { Dark = Colors.White, Light = Colors.Black } },
    },
});
```

**Comment by pictos** [comment]
@brminnick C# markup does have access to internal methods, right? If so, can we create a Factory method for `AppThemeBinding` ? Or is there something already?

**Comment by egvijayanand** [comment]
> @brminnick C# markup does have access to internal methods, right? If so, can we create a Factory method for `AppThemeBinding` ? Or is there something already?

Yeah, it can even be in the Markup package as there's already an extension method available for AppThemeBinding (instance-based) but not for **Setter**. The return type has to be `BindingBase` and invoked directly.

**Comment by brminnick** [comment]
We have an extension method, [`.AppThemeBinding()`](https://learn.microsoft.com/en-us/dotnet/communitytoolkit/maui/markup/extensions/bindable-object-extensions#appthemebinding)

```cs
new Label().AppThemeBinding(Label.TextColorProperty, Colors.Black, Colors.White);
```

**Comment by egvijayanand** [comment]
> We have an extension method, [`.AppThemeBinding()`](https://learn.microsoft.com/en-us/dotnet/communitytoolkit/maui/markup/extensions/bindable-object-extensions#appthemebinding)

Yes, but won't work in the context of **Setter** as there's no instance available to invoke this form of method. Style can be implicit also.

**Comment by egvijayanand** [comment]
@brminnick, it works seamlessly with the generic version `Style<T>`.

But one request, can another extension method be added to make it work with the non-generic version of Style to support the existing code base?

```cs
// Since Style is sealed, have defined as it is
public static Style AddAppThemeBinding(this Style style, BindableProperty property, object light, object dark)
{
    // Issue is AppThemeBinding is inaccessible due to its protection level
    style.Setters.Add(property, new AppThemeBinding() { Light = light, Dark = dark });
    return style;
}
```

**Comment by egvijayanand** [comment]
@brminnick Or to make it simple, add another constructor to generic `Style<T>` that takes a non-generic `Style` as a parameter. Any further operation after this is the same as working with that Style instance.

Since Style<T> can be implicitly converted to Style, the other way is not all an issue.

```cs
public Style(Style style) => MauiStyle = style;
```

**Comment by brminnick** [comment]
@egvijayanand I think that makes a lot of sense! 

Check out the proposal I just opened and leave a comment with your thoughts! 
https://github.com/CommunityToolkit/Maui.Markup/issues/274

**Comment by Ghostbird** [comment]
I found out that I can't access the `AppThemeBinding`. How do I make a multi-binding on text colour that also adapts to theme changes?
